magicgui/_qt/widgets/log_slider.py

```diff
@@ -0,0 +1,32 @@
+"""A Slider with a natural logarithmic scale."""
+import math
+
+from qtpy.QtCore import Qt
+from qtpy.QtWidgets import QSlider
+
+
+class QLogSlider(QSlider):
+    """A Slider Widget with a natural logarithmic scale."""
+
+    PRECISION = 1000
+
+    def __init__(self, parent=None):
+        super().__init__(Qt.Horizontal, parent=parent)
+        self.setMinimum(0)
+        self.setMaximum(10)
+
+    def value(self):
+        """Get the natural log slider value as a float."""
+        return math.log(super().value() / self.PRECISION)
+
+    def setValue(self, value):
+        """Set integer slier position from float ``value``."""
+        super().setValue(math.exp(value) * self.PRECISION)
```

Comment:
```suggestion
        super().setValue(int(math.exp(value) * self.PRECISION))
```
getting new warnings in the tests: `Implicit conversion to integers using __int__ is deprecated, and may be removed in a future version of Python.`

... and actually can you do the same to  `QDoubleSlider.setValue` and `setMaximum` while we're at it?